### DISCUSSION TITLE
Deep linking stopped working

### DISCUSSION THREAD

**Discussion by felixwanders** [Milad-Akarie/auto_route_library#1896] [General]
Hello everyone!

I'm not sure what I'm doing wrong, but it seems like deep linking stopped working in my app.
When testing deep linking with `adb` or `xcrun`, the app opens but it doesn't route to the desired page.

I'm using `auto_route` with nested routes and an `AutoTabsScaffold`.

Am I the only one in that case?

Thank you, have a good day 🌻 

>`pubspec.yaml`
>```yaml
>dependencies:
>  auto_route: ^7.9.2
>dev_dependencies:
> auto_route_generator: ^7.3.2
>```

**Comment by Milad-Akarie** [comment]
@omnitrogen are you sure you're on  auto_route: ^7.9.2?

**Reply by felixwanders** [reply]
Thank you for answering! I'm using version `7.9.2` indeed.

I'm working on a minimal reproducible example to understand if I'm doing something wrong.

I'll keep you updated!

Have a good day/night ☀️

**Reply by felixwanders** [reply]
Hey @Milad-Akarie, I've made a small repo with a minimal reproducible example of my issue with deep linking.
If you have time, you can have a look at it [here](https://github.com/omnitrogen/auto_route_deep_linking).
Thank you 🙂

**Comment by Milad-Akarie** [comment]
Hey @omnitrogen have you by any chance changed your deep-link scheme?

**Reply by felixwanders** [reply]
Nope, I haven't changed the deep-link scheme since I set it up.

**Comment by Milad-Akarie** [comment]
@omnitrogen what does your actual shame look like?

**Reply by felixwanders** [reply]
On [the minimal reproducible example](https://github.com/omnitrogen/auto_route_deep_linking) I've made, it looks like that : `deeplinking://router/page2`, which is the same format as the one I use in my app.

It is defined like so in `AndroidManifest.xml`:
```xml
<meta-data android:name="flutter_deeplinking_enabled" android:value="true"
<intent-filter>
    <action android:name="android.intent.action.VIEW"
    <category android:name="android.intent.category.DEFAULT" />
    <category android:name="android.intent.category.BROWSABLE" />
    <data android:scheme="deeplinking" android:host="router" />
</intent-filter>
```
and `Infos.plist`:

```xml
<key>CFBundleURLTypes</key>
<array>
	<dict>
		<key>CFBundleTypeRole</key>
		<string>Editor</string>
		<key>CFBundleURLName</key>
		<string>router</string>
		<key>CFBundleURLSchemes</key>
		<array>
			<string>deeplinking</string>
		</array>
	</dict>
</array>
<key>FlutterDeepLinkingEnabled</key>
<true />
```

**Comment by Milad-Akarie** [comment]
@omnitrogen  in the repo the scheme should be `deeplinking://router/page2`, in current and older versions of auto-route `router` is considered a part of the path as it does not get recognized as a valid hostname, that was a failed attempt by me to handle hostless deep-links because some people complained that the flutter upgrade broke deep-linking in auto_route, but obviously, I haven't given it  alot of thought then, what's confusing me right now is that was months ago...
<img width="752" alt="image" src="https://github.com/Milad-Akarie/auto_route_library/assets/55059449/d341c9a8-aed4-48d9-ae66-6a5013ee5d7a">
anyways, the next update will revert that fix, and hostless deep-links will have to be handled by the users.

**Reply by felixwanders** [reply]
Alright! That's great news, I'll wait for the next update then 🙂 
Thank you for your time!

**Comment by Milad-Akarie** [comment]
@omnitrogen can you confirm the cause of the issue, in your example change the host name router to something like router.dev and see if works?

**Reply by felixwanders** [reply]
I do confirm!
I just changed the host name to `router.dev` and deep linking to a nested tab route works!
Seems like it is the cause indeed 🎯

**Comment by felixwanders** [comment]
Fixed with version [`8.0.0`](https://pub.dev/packages/auto_route/changelog#800), thanks a lot @Milad-Akarie! 🥳
